app/models/miq_queue.rb

```diff
@@ -206,6 +209,20 @@ def self.submit_job(options)
     put(options)
   end
 
+  # TODO: Array param not seen yet
+  # @param resource [ExtManagementSystem|Nil|Array<Class,id>] resource for affinity. Typically an ems
+  def self.resource_enabled?(resource)
+    if resource.respond_to?(:ext_management_system)
```

Comment:
seems like this is missing a branch, where resource == ems?

Reply:
if resource == ems, then ```ems.ext_management_system``` == ```self```

Reply:
ah, ok, you are right, that is strange :-)